ls1intum/Artemis
# `General`: Check missing release date when creating or updating exercise

### Checklist
#### General


#### Client

### Motivation and Context


Previously only programming exercises warned the instructor that it does not have a release date. This behavior was inconsistent and dangerous. Additionally it was checked for test courses where this check is not necessary.

### Description

It is now checked for programming, modeling, file upload and text exercises and only in non-test courses.
If the exercise does not have a release date and is only updated, there is also no warning.

### Steps for Testing

Prerequisites:
- 1 Instructor

1. Log in to Artemis
2. Try to create an exercise (at least programming and one of the others). If you did not specify a release date, a warning should be shown.
3. Try to update the exercise. You should only get warned when you remove an existing release date
4. Check that the warning is never shown in a test course

#### Exam Mode Testing



Prerequisites:
- 1 Instructor

1. Log in to Artemis
2. Try to create an exercise, there should be no warning about a release date

### Review Progress






#### Code Review
#### Manual Tests
#### Exam Mode Test

### Test Coverage





### Screenshots


![Bildschirm­foto 2022-11-27 um 17 36 02](https://user-images.githubusercontent.com/38322605/204149737-f5144f44-f945-4d86-86b8-57c4f31331c5.png)
